code/engine/camera.lua

```diff
@@ -1,25 +1,285 @@
+--[[
+  PIXEL COORDINATES = love.mouse.getPosition() returns pixel coordinates
+  top-left = (0, 0), bottom-right = screen resolution, e.g. 1920x1080 or 3440x1440 and so on.
+
+  VIRTUAL RESOLUTION = Screen resolution / Scale. e.g. 1920x1080 / 6 = 320x180.
+  We're dealing with small images, and scaling the canvas up to make the images look larger than it actually is.
+
+  WORLD COORDINATES = Is the position of a entity or thing in the game world.
+  SCREEN COORDINATES = Pixel coordinates on the screen that is upscaled same as the images.
+]]
+
+ZOOM_MIN = -3
+ZOOM_MAX = 0
+ZOOM_ANIMATION_STEP = 1
+ZOOM_ANIMATION_SPEED = 0.0625
+ZOOM_ANIMATION_STATE = {
+  NONE = 0,
+  IN = 1,
+  OUT = 2,
+}
+
 local camera = {
-  --[[
-    resolution can be changed to
-    GAME_WIDTH and GAME_HEIGHT globals
-    ]]
-  visual_resolution_x = 384,
-  visual_resolution_y = 208,
-  translate_position_x = 0,
-  translate_position_y = 0,
+  x = 0,
+  y = 0,
+  scale = 6,
+  canvas_game = {},
+  canvas_hud = {},
+  delta_time = 0,
+  is_fullscreen = false,
+  follow_targets = {},
+  zoom = 0,
+  zoom_animation_coroutine = nil,
 }
+camera.__index = camera
+
+-- Get screen size with scale and zoom
+function camera:get_screen_game_size()
+  local width, height = love.graphics.getWidth(), love.graphics.getHeight()
+
+  return width / (self.scale + self.zoom), height / (self.scale + self.zoom)
+end
+
+function camera:get_screen_game_half_size()
+  local width, height = self:get_screen_game_size()
+
+  return width / 2, height / 2
+end
+
+function camera:get_aspect_ratio()
+  local width, height = self:get_screen_game_size()
+
+  return width / height
+end
+
+-- Returns scale diff between game and hud
+function camera:get_screen_game_hud_diff() return self.scale / (self.scale + self.zoom) end
```

Comment:
Namnet känns lite missvisande iom det inte returnerar en skillnad utan typ... förhållande eller så? Osäker på hur HUD kommer in?

Reply:
döpte om denna till get_zoom_aspect_ratio